cupy/core/core.pyx

```diff
@@ -2856,14 +2866,15 @@ cpdef ndarray tensordot_core(
                 b.data.ptr, runtime.CUDA_R_16F, <int>ldb,
                 a.data.ptr, runtime.CUDA_R_16F, <int>lda,
                 <size_t>&zero_fp32,
-                c.data.ptr, Ctype, <int>m,
+                c.data.ptr, runtime.CUDA_R_16F, <int>m,
```

Comment:
Do we know for sure `c` is of type `float16` at this stage?

Reply:
Yes, I checked the original code, and if dtype of matrix `a` and `b` is `float16`, then the dtype of matrix `c` will be always `float16`.

Reply:
Thanks!